extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java

```diff
@@ -96,6 +102,28 @@ VertxWebRouterBuildItem initializeRouter(VertxHttpRecorder recorder,
         return new VertxWebRouterBuildItem(router);
     }
 
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    void initializeAdditionalStaticResource(StaticResourceRecorder recorder, BuildProducer<RouteBuildItem> routes,
+            StaticResourceBuildTimeConfig staticResourceBuildTimeConfig,
+            StaticResourceRuntimeConfig staticResourceRuntimeConfig,
+            BuildProducer<NotFoundPageDisplayableEndpointBuildItem> displayableEndpoints) {
+        if (staticResourceBuildTimeConfig.isEnabled()) {
+            final String staticResourceEndpoint = getStaticResourceEndpoint(staticResourceBuildTimeConfig);
+            Handler<RoutingContext> handler = recorder.handler(staticResourceBuildTimeConfig, staticResourceRuntimeConfig,
+                    staticResourceEndpoint);
+            routes.produce(new RouteBuildItem(staticResourceEndpoint, handler));
+            routes.produce(new RouteBuildItem(staticResourceEndpoint + "/*", handler));
+            displayableEndpoints.produce(new NotFoundPageDisplayableEndpointBuildItem(staticResourceEndpoint + "/"));
+        }
+    }
+
+    private String getStaticResourceEndpoint(StaticResourceBuildTimeConfig staticResourceBuildTimeConfig) {
+        String endpoint = staticResourceBuildTimeConfig.endpoint;
```

Comment:
Right now, I think if you define `/`as the path, it will work and it definitely should be absolutely avoided and throw a `ConfigurationException`.

Reply:
Good point.